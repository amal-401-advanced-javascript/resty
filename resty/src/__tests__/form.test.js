import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Form from '../components/form/form.js';


describe('<Form/>', () => {
    it('is alive at application start', () => {
      const form = shallow(<Form />);
      expect(form.find('main').exists()).toBeTruthy();
    });
    it('method GET', () => {
      const form = mount(<Form />);
      const button = form.find('#get');
      button.simulate('click');
      expect(form.state('method')).toBe('GET');
    });
    it('method POST', () => {
        const form = mount(<Form />);
        const button = form.find('#post');
        button.simulate('click');
        expect(form.state('method')).toBe('POST');
      });
      it('method PUT', () => {
        const form = mount(<Form />);
        const button = form.find('#put');
        button.simulate('click');
        expect(form.state('method')).toBe('PUT');
      });
      it('method DELETE', () => {
        const form = mount(<Form />);
        const button = form.find('#delete');
        button.simulate('click');
        expect(form.state('method')).toBe('DELETE');
      });
    it('Renders correctly', () => {
      const tree = renderer.create(<Form />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });