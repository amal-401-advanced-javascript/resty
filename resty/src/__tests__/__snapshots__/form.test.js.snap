// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> Renders correctly 1`] = `
<main>
  <div
    className="methodDiv"
  >
    <label
      className="url"
    >
      URL:
    </label>
    <input
      className="url"
      onChange={[Function]}
      type="text"
    />
    <button
      className="go"
      id="go"
      onClick={[Function]}
    >
      Go!
    </button>
    <br />
    <button
      className="method"
      id="get"
      onClick={[Function]}
    >
      GET
    </button>
    <button
      className="method"
      id="post"
      onClick={[Function]}
    >
      POST
    </button>
    <button
      className="method"
      id="put"
      onClick={[Function]}
    >
      PUT
    </button>
    <button
      className="method"
      id="delete"
      onClick={[Function]}
    >
      DELETE
    </button>
  </div>
  <div
    className="output"
  >
    
     
  </div>
</main>
`;
